src/main/java/spring/intro/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package spring.intro.service.impl;
+
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import spring.intro.dao.UserDao;
+import spring.intro.model.User;
+import spring.intro.service.UserService;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    @Autowired
+    @Qualifier(value = "userDaoImpl")
```

Comment:
Do you need a qualifier now?

Reply:
No we need not, because in qualifier define concrete implementaion.